app-modules/report/src/Filament/Widgets/ResearchAdvisorReportStats.php

```diff
@@ -0,0 +1,136 @@
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\Report\Filament\Widgets;
+
+use AdvisingApp\Research\Models\ResearchRequest;
+use Filament\Widgets\StatsOverviewWidget\Stat;
+use Illuminate\Contracts\Database\Eloquent\Builder;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Number;
+
+class ResearchAdvisorReportStats extends StatsOverviewReportWidget
+{
+    protected int | string | array $columnSpan = 'full';
+
+    public function getStats(): array
+    {
+        $startDate = $this->getStartDate();
+        $endDate = $this->getEndDate();
+
+        $shouldBypassCache = filled($startDate) || filled($endDate);
+
+        $researchAdvisorsCount = $shouldBypassCache
+            ? ResearchRequest::query()
+                ->whereNotNull('title')
+                ->when(
+                    $startDate && $endDate,
+                    fn (Builder $query): Builder => $query->whereBetween('created_at', [$startDate, $endDate])
+                )
+                ->count()
+            : Cache::tags(["{{$this->cacheTag}}"])->remember(
+                'research-advisors-count',
+                now()->addHours(24),
+                fn (): int => ResearchRequest::query()
+                    ->whereNotNull('title')
+                    ->count(),
+            );
+
+        $uniqueUsersCount = $shouldBypassCache
+            ? ResearchRequest::query()
+                ->whereNotNull('title')
+                ->when(
+                    $startDate && $endDate,
+                    fn (Builder $query): Builder => $query->whereBetween('created_at', [$startDate, $endDate])
+                )
+                ->distinct('user_id')
+                ->count('user_id')
+            : Cache::tags(["{{$this->cacheTag}}"])->remember(
+                'research-advisor-unique-users-count',
+                now()->addHours(24),
+                fn (): int => ResearchRequest::query()
+                    ->whereNotNull('title')
+                    ->distinct('user_id')
+                    ->count('user_id'),
+            );
+
+        $sourcesCount = $shouldBypassCache
```

Comment:
I think you missed the sources that are defined in the `sources` JSON column of the request too, those also need to be counted